### ISSUE TITLE
Add gzip option

### ISSUE THREAD

**Issue by chiefcll** [Polymer/polymer-build#195]
I'd like for builds to include a gzip option

"builds": [{
    "name": "bundled",
    "bundle": true,
    "js": {"minify": true},
    "css": {"minify": true},
    "html": {"minify": true},
    "gzip": {"suffix": ".gz"},
  }],

Adding such would run the files through gzip and output bundle.js.gz assets. This would save everyone from having to gzip as an additional step.

**Comment by web-padawan** [comment]
Why not to delegate this to `nginx` or other production server? IMO, `polymer-buildi` is for polymer-specific magic. Maybe `polymer-cli` could handle this, although that would make it even larget than it is.

**Comment by chiefcll** [comment]
We have in the past and ran into issues with nginx returning 0 as the file size. Rather than having the server gzip files on the fly we prefer to upload them already gzipped and serve them that way. 

I guess a more generic option would be - can we add a middleware into polymer-build?

**Comment by stramel** [comment]
You could create a simple gulpfile that just adds compression and wraps polymer-build

**Comment by chiefcll** [comment]
Yes - but we also use polymer-cli - so now we need all custom tooling. Multiply that by however many users all having to add another layer to gzip files, which I think is pretty standard practice for web servers today to serve gzipped files. If you're already doing the work to minify, uglify, concatenate - why not also gzip?

**Comment by stramel** [comment]
I'm not seeing where you need all custom tooling. Just instead of running `polymer build` you would run `gulp`/`gulp build` or alias in package.json to be `npm run build` (or `yarn build` if you are using yarn instead of npm). The rest of your commands would remain the same, `init`, `lint`, etc.

You are replacing the build command. I assume the polymer team is going to say that it would be custom added or done on the webserver. But maybe @rictic @FredKSchott or @usergenic  could speak to it?

**Comment by chiefcll** [comment]
Running `polymer build` is much simpler than writing my own build step to replace that (I have multiple times in the past) - I'm trying to use the tool as is (which works pretty well). Again, I may be one of thousands of users who probably want to gzip the files before giving them to the server - Then the question becomes should we all be writing a custom build step to do that?

**Comment by chiefcll** [comment]
Follow up - I am using the polymer-cli to do `polymer build` - Should I pass this issue up to that repo?
